src/test/java/io/kestra/plugin/googleworkspace/calendar/EventCreatedTriggerTest.java

```diff
@@ -0,0 +1,223 @@
+            .setHtmlLink("https://calendar.google.com/event?eid=eventSomething")
+            .setCreated(new DateTime(System.currentTimeMillis()))
+            .setUpdated(new DateTime(System.currentTimeMillis()));
+
+        // Add start and end times
+        com.google.api.services.calendar.model.EventDateTime start = 
+            new com.google.api.services.calendar.model.EventDateTime()
+                .setDateTime(new DateTime(System.currentTimeMillis() + 3600000)) // In 1 hour
+                .setTimeZone("UTC");
+        
+        com.google.api.services.calendar.model.EventDateTime end = 
+            new com.google.api.services.calendar.model.EventDateTime()
+                .setDateTime(new DateTime(System.currentTimeMillis() + 7200000)) // in 2 hours
+                .setTimeZone("UTC");
+
+        googleEvent.setStart(start);
+        googleEvent.setEnd(end);
+
+        // Add organizer
+        Event.Organizer organizer = new Event.Organizer()
+            .setEmail("lethal@company.com")
+            .setDisplayName("Meeting Organizer");
+        googleEvent.setOrganizer(organizer);
+
+        EventCreatedTrigger.EventMetadata metadata = trigger.convertToEventMetadata(googleEvent);
+
+        assertThat(metadata.getId(), equalTo("eventSomething"));
+        assertThat(metadata.getSummary(), equalTo("Test Meeting"));
+        assertThat(metadata.getDescription(), equalTo("This is a test meeting"));
+        assertThat(metadata.getLocation(), equalTo("Conference of no Return"));
+        assertThat(metadata.getStatus(), equalTo("confirmed"));
+        assertThat(metadata.getHtmlLink(), equalTo("https://calendar.google.com/event?eid=eventSomething"));
+        assertThat(metadata.getOrganizer().getEmail(), equalTo("lethal@company.com"));
+        assertThat(metadata.getOrganizer().getDisplayName(), equalTo("Meeting Organizer"));
+        assertThat(metadata.getStart().getTimeZone(), equalTo("UTC"));
+        assertThat(metadata.getEnd().getTimeZone(), equalTo("UTC"));
+    }
+
+    @Test
+    @DisabledIf("isServiceAccountNotExists")
+    void shouldConnectToCalendarAPI() throws Exception {
+        var trigger = EventCreatedTrigger.builder()
+            .id(IdUtils.create())
+            .serviceAccount(Property.ofValue(serviceAccount()))
+            .calendarId(Property.ofValue("primary"))
+            .interval(Duration.ofMinutes(5))
+            .build();
+
+        RunContext runContext = runContextFactory.of(Map.of());
+
+        // This should succeed and create a connection
+        com.google.api.services.calendar.Calendar calendar = trigger.connection(runContext);
+        
+        assertThat(calendar, notNullValue());
+        assertThat(calendar.getApplicationName(), equalTo("Kestra"));
+    }
+
+    private static boolean isServiceAccountNotExists() {
+        return serviceAccount() == null;
+    }
+
+    private static String serviceAccount() {
+        return System.getenv("GOOGLE_SERVICE_ACCOUNT_JSON");
```

Comment:
In `.github/setup-unit.sh` it's like this:

```bash
echo $GOOGLE_SERVICE_ACCOUNT | base64 -d > src/test/resources/.gcp-service-account.json
```

So:


```suggestion
        return System.getenv("GOOGLE_SERVICE_ACCOUNT");
```